spec/reek/smells/instance_variable_assumption_spec.rb

```diff
@@ -0,0 +1,184 @@
+require_relative '../../spec_helper'
+require_lib 'reek/smells/instance_variable_assumption'
+
+RSpec.describe Reek::Smells::InstanceVariableAssumption do
+  context 'report' do
+    context 'smell line' do
+      it 'should report the core class in the message' do
+        src = <<-EOS
+          class Dummy
+            def test
+              @a = 1
+            end
+          end
+        EOS
+
+        expect(src).to reek_of(:InstanceVariableAssumption, lines: [1])
+      end
+    end
+
+    context 'smell context' do
+      it 'should report the class as the context' do
+        src = <<-EOS
+          class Dummy
+            def test
+              @a = 1
+            end
+          end
+        EOS
+
+        expect(src).to reek_of(:InstanceVariableAssumption, context: 'Dummy')
+      end
+    end
+
+    context 'smell message' do
+      it 'should report the class and ivars in the message' do
```

Comment:
I'm not sure this one should smell, since the only use here is an actual assignment, so it doesn't assume the variable is set.